src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -90,11 +94,29 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String user = userService.getCurrentUser().getEmail();
     String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
     String recipient = request.getParameter("recipient");
+    System.out.println("before getting sentiment score");
+    System.out.println("text: " + text);
+    float sentimentScore = calculateSentimentScore(text);
```

Comment:
This isn't really a technical comment so much as a design comment...

It looks like you are putting sentiment analysis in the critical path of posting a message, which may delay the operation of posting a message (and also make it more likely for uploading a message to fail).

Please use the App Engine task queue functionality:
https://cloud.google.com/appengine/docs/standard/java/taskqueue/push/creating-push-queues

Or the more general Cloud PubSub mechanism:
https://cloud.google.com/pubsub/

To schedule sentiment analysis _after_ successfully uploading the message (you may also want to schedule periodic recalculation of sentiment analysis to catch any existing messages and also to keep the sentiment analysis scoring up-to-date with changes/improvements in the sentiment scoring algorithm).